src/common/App.js

```diff
@@ -9,13 +10,14 @@ class App extends Component {
                     React SSR and Router v4 example.
                 </p>
                 <hr />
-                <Header>
-                    <h1>Search results</h1>
-                </Header>
-                <SearchResults queryData={this.props.data} />
+                <Switch>
+                  <Route exact path="/about" render={() => <About/>} />

```

Comment:
As of current version of React & React Router I can see the syntax `render={() => <About/>}` does not throw exceptions in console. It's good, however please note that in early versions of React v.16, framework was complaining about incorrect use of HTML attributes (standard HTML rather than JSX). I believe now it has been improved & fixed.

If you ever encounter it, don't waste your time on debugging, since I was trying to go down this path and it simply didn't work. React upgrade apparently did the job. Good!